benchadapt/python/benchadapt/adapters/_adapter.py

```diff
@@ -17,16 +17,42 @@ class BenchmarkAdapter(abc.ABC):
     command : List[str]
         A list of args to be run on the command line, as would be passed
         to `subprocess.run()`.
+    result_defaults_override : Dict[str, Any]
+        A dict of default values to be passed to `BenchmarkResult`. Useful for
+        specifying metadata only available at runtime, e.g. build info. Overrides
+        `BenchmarkResult` defaults; is overridden by values passed in
+        ``transform_results()``. Values of ``None`` do not unset defaults, just as
+        when passed to ``BenchmarkResult`'s init method.
+    results_defaults_append : Dict[str, Any]
+        A dict of default values to be appended to `BenchmarkResult` values.
+        Appended after instantiation. Useful for appending extra tags or other
+        metadata in addition to that gathered elsewhere. Only applicable for dict
+        attributes. For each element, will override any keys that already exist,
+        i.e. it does not append recursively.
```

Comment:
hm, so even after reading the docs, source, and tests, I still don't have a good intutition for everything that will be put into these defaults. Could you add concrete examples to the docs? Like "put cluster info and XYZ here" and "put extra tags here"? We can always modify the docs later if the status quo changes.

Having said that, I wonder if it would be better to scope these new variables to just modifying cluster info and tags if that's all we need to modify right now. Unless there will be other attributes to set defaults for, and you worry it'd get out of control? As this PR stands, these new attributes seem too generic to me to reason about.

Reply:
Yeah, if you anticipate putting all 6 of those fields in one of these two `result_defaults_` dicts in future implementations, then that would be way too many to first-class here. I'm sure I'll understand better once I see the Arrow adapter implementation.

Reply:
Kept this UI, aside from renaming `result_defaults_*` to `result_fields_*` (since they're now applied after, not as defaults). Still not sure I love this UI (or the name), but it is powerful and easy to maintain. We could add some syntactic sugar methods to build the dicts after init, though I'm not really sure that's better, and it would make it easier to call things in the wrong order and end up in a weird state.

Reply:
I like this!